### ISSUE TITLE
Implement data download bundle

### ISSUE THREAD

**Issue by jmcarp** [18F/openFEC#1383]
As described in https://github.com/18F/openFEC-web-app/issues/946#issuecomment-161409921, we're going to package downloads in a compressed file with a manifest, so that users can understand what's in their downloads after they've downloaded them. The bundle can include a manifest, which will explain the bundle's contents, and potentially a data dictionary as well.

* [x] Create bundle
* [x] Add manifest
* [x] Compress
* [ ] Add data dictionary

I can start working on the basics now, but we'll want to discuss the contents of the bundle in more detail before this is released. What should go into the manifest? What compression format should we use?

Ping @jenniferthibault @noahmanger @LindsayYoung

**Comment by noahmanger** [comment]
Off the top of my head, I think the bundle should just include the .csv file and a .txt manifest, which would contain:
- Timestamp of the download
- Type of resource
- Filters applied
- Total number of records
- Data dictionary for all the fields

Can anyone think of anything else that would be useful?

Also, I noticed that the .csv's currently have their columns organized alphabetically, but this means that in the case of candidates or committees, the actual name of each is not the first column. Do we want to organize these in a more logical way so that the main identifier (name or id) is first? I don't know.

**Comment by jenniferthibault** [comment]
I would say yes absolutely we'd want to reorganize the .csv columns in a more logical way if that's at all possible. 

I _think_ your list about covers it @noahmanger , as long as the timestamp includes the date.

**Comment by LindsayYoung** [comment]
I like the idea of having the columns in a user friendly format, but I don't think we need to do that right away.

**Comment by jmcarp** [comment]
By the way, the timestamp in the manifest isn't guaranteed to be the same as the timestamp the user sees on the data site. If multiple users ask for the same file at different times, they'll each see the time they asked for the file on the site, but the manifest would have to contain the timestamp we actually got the request. Does that make sense?

**Comment by noahmanger** [comment]
That makes sense. We should just be clear. Oh, could we separately include the last time the data was updated? I think that that's actually the most important time point to capture.

**Comment by LindsayYoung** [comment]
In the downloads, you should already have the dates applicable to the records and when the forms were filed in most cases. 

People might get confused if we say when our database was updated, because the processes the data form the day before and even if we showed when the database was updated, how recent the data is depends on the filing schedule of the committee. Then, when you look at data from multiple committees with different reporting structures, looking at filing schedules makes less sense.

**Comment by noahmanger** [comment]
Added a story card here: https://18f.storiesonboard.com/m/fec/!card/QfobeIdNeNoMhDnUfLBCIg

**Comment by jmcarp** [comment]
I'm thinking the data dictionary may be a separate feature. We can leverage our existing Swagger documentation for the query parameters, but we haven't actually specified the documentation for the model fields. In other words, we have to add the docs before we can use them to build data dictionaries. We started this a while back in #1036 but never finished. If we want to include data dictionaries with the first release of data downloads, we can get back to documentation now.

**Comment by noahmanger** [comment]
So I made a [separate issue](https://github.com/18F/openFEC/issues/1416) for the data dictionary, which can go with a future release. Does the manifest currently at least have filter definitions and everything else listed above?
